homeassistant/components/minecraft_server/config_flow.py

```diff
@@ -0,0 +1,91 @@
+"""Config flow for Minecraft Server integration."""
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigFlow
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PORT, CONF_SCAN_INTERVAL
+
+from . import MinecraftServer
+from .const import (  # pylint: disable=unused-import
+    DEFAULT_HOST,
+    DEFAULT_NAME,
+    DEFAULT_PORT,
+    DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+)
+
+
+class MinecraftServerConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Minecraft Server."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        if user_input is None:
+            # No configuration data available yet, show default configuration form.
+            return await self._show_config_form()
+
+        # User inputs.
+        host = user_input[CONF_HOST]
+        port = user_input[CONF_PORT]
+        scan_interval = user_input[CONF_SCAN_INTERVAL]
+
+        # Abort in case the host was already configured before.
+        unique_id = f"{host.lower()}-{port}"
+        await self.async_set_unique_id(unique_id)
+        self._abort_if_unique_id_configured()
+
+        errors = {}
+
+        # Validate port configuration (limit to user and dynamic port range).
+        if (port < 1024) or (port > 65535):
```

Comment:
Do these checks as part of the data voluptuous schema. Use `vol.Range`.

Reply:
Initially I tried to do it as part of the schema, but then the config flow UI showed a slider instead of a number input field, which I didn't like. Do you have an idea how I can use `vol.Range` but without having the slider afterwards in the UI?

Reply:
What did you try? Did you try with `vol.All`?

Reply:
I tried the following and each time the slider was shown:
`vol.All(vol.Coerce(int), vol.Range(min=1024, max=65535))`
`vol.All(int, vol.Range(min=1024, max=65535))`

And without vol.All and don't see any port input in the config flow form at all:
`vol.Range(min=1024, max=65535)`

Reply:
Ok.